### PR TITLE
Improve extremely inefficient darkness-checking logic.

### PR DESCRIPTION
**Pull request by dbrant**
It looks like you have a function that checks whether the current image is "too dark", based on the fraction of pixels that fall below a certain threshold.

Nothing wrong with that; however: this function seems to be operating on the *entire* full-resolution bitmap, which makes it horribly inefficient. On my device, this process was taking as long as **three seconds**.

Here's the thing:  If you think about it, the darkness checking can be performed on the *thumbnail* of the image, since the thumbnail will contain the same fraction of dark pixels as the original bitmap.  The checking can therefore be performed in a tiny fraction of the time.

The new method takes less than 100 milliseconds, instead of several seconds.

### CODE REVIEW

**Comment by domdomegg** [comment]
To avoid adding a library, rather than generate a thumbnail could we just sample some number of pixels for each dimension - e.g. take pixels at positions `[x * width/100, y * height/100]` for `x = [0..99]`, `y = [0..99]`?

**Comment by dbrant** [comment]
@domdomegg This isn't generating a thumbnail; it's retrieving the pre-made Exif thumbnail that is already baked into the image. It never touches the primary bitmap.

**Comment by domdomegg** [comment]
> @domdomegg This isn't generating a thumbnail; it's retrieving the pre-made Exif thumbnail that is already baked into the image. It never touches the primary bitmap.

Ah makes sense. Is it possible that the exif thumbnail won't exist in that case?

**Comment by dbrant** [comment]
It gets included by all the devices I've ever encountered.  And if it doesn't, then the logic will just fall back to the old way.

Another note: adding `androidx.exifinterface` isn't really adding a new library, but rather making correct use of AndroidX.  Since there are several other points in the app where you make use of `ExifInterface`, it is now recommended to use AndroidX for those, as well. (Android Studio shows a lint warning for it.)

**Comment by nicolas-raoul** [comment]
Here are 4 images that are good tests for the detection algorithm: #1767

Making a unit test that uses these 4 images could be a nice little task for newcomers to take :-)

**Comment by domdomegg** [comment]
> Here are 4 images that are good tests for the detection algorithm: #1767
> 
> Making a unit test that uses these 4 images could be a nice little task for newcomers to take :-)

Have opened #2649

File: app/src/main/java/fr/free/nrw/commons/utils/ImageUtils.java [hunk 1]
```diff
@@ -83,33 +84,26 @@
...
-        int checkImageTopPosition = 0;
-        int checkImageLeftPosition = 0;
-
-        Timber.v("left: " + checkImageLeftPosition + " right: " + loadImageWidth + " top: " + checkImageTopPosition + " bottom: " + loadImageHeight);
-
-        Rect rect = new Rect(checkImageLeftPosition,checkImageTopPosition, loadImageWidth, loadImageHeight);
+    static @Result int checkIfImageIsTooDark(String imagePath) {
+        long millis = System.currentTimeMillis();
+        try {
+            Bitmap bmp = new ExifInterface(imagePath).getThumbnailBitmap();
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/utils/ImageUtils.java]: While testing the PR, lets check if the thumbnail information is present in EXIF for images picked from Google photos.

**Comment by dbrant** [comment]
I've tried and verified the feature with the images linked above, as well as a few others (including some without an Exif thumbnail).

I assume the unit test(s) for this feature can be done in a separate PR?

**Comment by nicolas-raoul** [comment]
Yes the tests will be another PR 🙂

**Pull request merged**